docs/staging-use-cases.md

```diff
@@ -0,0 +1,209 @@
+---
+
+# Cluster API Reference Use Cases
+
+This is a living document that serves as a reference and a staging area for use cases collected from the community during post-v1alpha1 project redesign.
+
+## Table of Contents
+
+* [Reference Use Cases](#cluster-api-reference-use-cases)
+  * [Metadata](#metadata)
+  * [Table of Contents](#table-of-contents)
+  * [Role Glossary](#role-glossary)
+  * [Icon Glossary](#icon-glossary)
+  * [Operator of Workload Cluster](#operator-of-workload-cluster)
+  * [Operator of Management Cluster](#operator-of-management-cluster)
+  * [Multi-Cluster/Multi-Provider](#multi-cluster/multi-provider)
+  * [Provider Implementors](#provider-implementors)
+
+## Role Glossary
+- __User__: consumer of a Kubernetes-conformant cluster created by the Cluster API.
+    - Does not use Cluster API.
+- __Operator__: Administrator responsible for creating and managing a Kubernetes cluster deployed by Cluster API.
+    - Uses Cluster API.
+- __Multi-cluster operator__: An operator responsible for multiple Kubernetes clusters deployed by Cluster API.
+    - Uses Cluster API.
+    - Cares about keeping config similar between many clusters.
+
+
+## Icon Glossary
+- 🔭 Out of Scope for Cluster API itself, but should be possible via higher level tool.
+    - These are use-cases that we should take care not to prevent.
+
+## Operator of Workload Cluster
+
+### Creating Clusters
+
+- As an operator, given that I have a cluster running Cluster API, I want to be able to use declarative APIs to manage another Kubernetes cluster (create, upgrade, scale, delete).
+
+- As an operator, given that I have a cluster running Cluster API, I want to be able to use declarative APIs to manage a vendor’s Kubernetes conformant cluster (create, upgrade, scale, delete).
+
+- As an operator, when I create a new cluster using Cluster API, I want Cluster API to automatically create and manage any supporting provider infrastructure needed for my new cluster to function.
+
+- As an operator, when I create a new cluster using Cluster API, I want to be able to use existing infrastructure (e.g. VPC’s, SecurityGroups, veth, GPUs).
+
+- 🔭 As an operator, when I create a new cluster using Cluster API, I want to be able to take advantage of resource aware topology (e.g. compute, device availability, etc.) to place machines.
+
+- As an operator, I need to have a way to make minor customizations before kubelet starts while using a standard node image and standard boot script. Example: write a file, disable hyperthreading, load a kernel module.
+
+- As an operator, I need to have a way to apply labels to Nodes created through ClusterAPI. This will allow me to partition workloads across Nodes/Machines and MachineDeployments. Examples of labels include datacenter, subnet, and hypervisor, which applications can use in affinity policies.
+
+- As an operator, I want to be able to provision the nodes of a workload cluster on an existing vnet that I don’t have admin control of.
+
```

Comment:
As an operator, if there are no mandatory provider-specific fields, it should be possible to create a cluster by only specifying the provider (i.e. there should be defaults for all required upstream Cluster API types).